File: client/src/types/Order.ts
export interface Order {
  orderId: string;
  customerId: string;
  status: OrderStatus;
  createdAt: string;
  totalAmount: number;
  notes?: string;
  items: OrderItem[];
}

export interface OrderItem {
  productId: string;
  productName: string;
  quantity: number;
  price: number;
  totalPrice: number;
}

export interface OrderSummary {
  orderId: string;
  status: string;
  createdAt: string;
  totalAmount: number;
  itemCount: number;
}

export interface CreateOrderRequest {
  customerId: string;
  items: CreateOrderItemRequest[];
  notes?: string;
}

export interface CreateOrderItemRequest {
  productId: string;
  productName: string;
  quantity: number;
  price: number;
}

export interface CreateOrderResponse {
  orderId: string;
  status: string;
  totalAmount: number;
  createdAt: string;
}

export enum OrderStatus {
  Pending = 'Pending',
  Confirmed = 'Confirmed', 
  Cancelled = 'Cancelled',
  Shipped = 'Shipped',
  Delivered = 'Delivered'
}

export const OrderStatusColors = {
  [OrderStatus.Pending]: 'orange',
  [OrderStatus.Confirmed]: 'green',
  [OrderStatus.Cancelled]: 'red',
  [OrderStatus.Shipped]: 'blue',
  [OrderStatus.Delivered]: 'purple'
};

File: OrderService/Domain/Entities/Order.cs
using System.ComponentModel.DataAnnotations;

namespace OrderService.Domain.Entities;

public class Order
{
    public Guid Id { get; set; }
    
    [Required]
    public Guid CustomerId { get; set; }
    
    public DateTime CreatedAt { get; set; }
    
    public OrderStatus Status { get; set; }
    
    public List<OrderItem> Items { get; set; } = new();
    
    public decimal TotalAmount { get; set; }
    
    public string? Notes { get; set; }
}

public enum OrderStatus
{
    Pending = 0,
    Confirmed = 1,
    Cancelled = 2,
    Shipped = 3,
    Delivered = 4
}